newspaper/mthreading.py

```diff
@@ -105,4 +104,7 @@ def set(self, paper_list, threads_per_source=1):
         self.pool = ThreadPool(num_threads, timeout)
 
         for paper in self.papers:
-            self.pool.add_task(paper.download_articles)
+            try:
+                self.pool.add_task(paper.download_articles)
+            except AttributeError:
```

Comment:
I guess long-term the clean solution would be to see `isinstance(object, Article) or isinstance(object, Source)` and call the appropriate method after. I'll add a TODO